webapp/channels/src/components/common/hooks/use_desktop_notification_permission.ts

```diff
@@ -0,0 +1,52 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+import {useCallback, useEffect, useState} from 'react';
+
+import type {NotificationPermissionNeverGranted} from 'utils/notifications';
+import {isNotificationAPISupported} from 'utils/notifications';
+import {isDesktopApp} from 'utils/user_agent';
+
+export type DesktopNotificationPermission = Exclude<NotificationPermission, typeof NotificationPermissionNeverGranted> | undefined;
+
+// We store the permission state globally here to avoid calling requestPermission() multiple times
+let desktopNotificationPermissionGlobalState: DesktopNotificationPermission | undefined;
```

Comment:
0/5, should we store this in view state? I try to avoid globals.

Reply:
In this case, I think a global is fine. I don't think it's worth the complexity of storing it in Redux since we're not likely to use it elsewhere without the hook